transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/GatewaySessionCtx.java

```diff
@@ -89,6 +90,9 @@ private void onDeviceConnect(String deviceName, String deviceType) {
         if (!devices.containsKey(deviceName)) {
             Optional<Device> deviceOpt = deviceService.findDeviceByTenantIdAndName(gateway.getTenantId(), deviceName);
             Device device = deviceOpt.orElseGet(() -> {
+                JsonNode forbidCreateDevice = gateway.getAdditionalInfo().get("forbid_create_device");
```

Comment:
please rename to "allowDeviceCreation" to be camel case. (The boolean condition need to be reversed).

Reply:
This boolean flag should be fetched. No need to get and calculate this each time the device sends a request.

Reply:
sorry, it is hard to find a suitable place to set this attribute for gateway device.